src/Providers/MapService.js

```diff
@@ -64,7 +64,9 @@ export var MapServiceProvider = MapService.extend({
         }
         for (var i = 0; i < features.features.length; i++) {
           var feature = features.features[i];
-          layer = (layer) ? layer : response.results[i].layerId;
+          // layer = (layer) ? layer : response.results[i].layerId;
+          layer = layer || response.results[i].layerId;
```

Comment:
i had to make this change manually and i'm not certain my replacement line is equivalent.